### DISCUSSION TITLE
Is there a correct way to create forms in bulk?

### DISCUSSION THREAD

**Discussion by Arbureva** [AllenDang/giu#564] [Q&A]
My current code uses `[]*g.TableRowWidget` to batch create and then reassign values, but in the process of using it I found a lot of problems, his ID or some kind of tag seems to be changed, is there a way to add rows in real time?

**Comment by Arbureva** [comment]
#542 I've used the way I've created it now here before, and while I say I've solved all the problems so far, I'm afraid I've solved them in some strange way that has led to some unknown bugs

**Reply by Arbureva** [reply]
I looked into the `giu.RangeBuilder()` method, but its `giu.Custom` does not apply to table rows. While it is possible that this is a marker to distinguish between start and end, what makes me curious is whether he would do this to prevent performance or some problems?

**Comment by gucio321** [comment]
to generate rows in real-time I usually make list of rows, fill the list in for-loop and in my builder, I just call `giu.Table().Rows(myRows...)`
Could you please show some demo?

**Reply by Arbureva** [reply]
Okay, I'll write a demo right away.

**Reply by Arbureva** [reply]
this => https://github.com/Locter9001/giu-demo

**Reply by Arbureva** [reply]
I will use it in the pop-up box this time, the normal method may not work, it seems that the content of the pop-up box is fixed when it is generated, and the layout cannot be changed.

**Reply by gucio321** [reply]
hmm, well, I see that something's going wron there...

**Reply by Arbureva** [reply]
I mean to write a `layout` variable in `giu.PopupModal` and assign the value to the variable externally

**Reply by gucio321** [reply]
okey, so what's wrong? do you mean this crazy scrollbar in your example?

**Reply by gucio321** [reply]
[Screencast from 10-06-2022 06:41:56 PM.webm](https://user-images.githubusercontent.com/73652197/194370739-5935868b-22f8-4fde-b67d-82171dd5e6e9.webm)

**Reply by Arbureva** [reply]
Yes, but you don't have a flickering screen here yet

**Reply by Arbureva** [reply]
My system is windows 11, the processor is intel i7 12700K, currently no discrete graphics card

**Reply by gucio321** [reply]
really strange...
I've modified the code this way:
```golang
package main

import (
	"time"

	"github.com/AllenDang/giu"
)

var (
	rows       []*giu.TableRowWidget
	personList []person
)

type person struct {
	name int
	age  int
}

func loop() {
	rows = rebuildRows()
	giu.SingleWindowWithMenuBar().Layout(
		giu.Table().Columns(
			giu.TableColumn("name"),
			giu.TableColumn("age"),
		).FastMode(false).Rows(
			// rows...,
			rebuildRows()...,
		),
	)
}

func main() {
	w := giu.NewMasterWindow("Overview", 200, 500, 0)
	go refresh()
	w.Run(loop)
}

func refresh() {
	ticker := time.NewTicker(time.Millisecond * 100)
	name := 10
	age := 18
	for {
		if name == 100 {
			break
		}
		person := person{
			name: name + 1,
			age:  age + 1,
		}

		personList = append(personList, person)

		name++
		age++
		<-ticker.C

		// rows = rebuildRows()
		// giu.Update()
	}
}

func rebuildRows() []*giu.TableRowWidget {
	var rowList []*giu.TableRowWidget
	for i := 0; i < len(personList); i++ {
		rowList = append(rowList, giu.TableRow(
			giu.Selectablef("%v", personList[i].name),
			giu.Selectablef("%v", personList[i].age),
		))
	}

	return rowList
}
```
It seems that as that as long as the list ist being generated, the window behaves reallly strang
when it is done, everything is ok...
CC @AllenDang

**Reply by Arbureva** [reply]
Yes, the screen will flicker and lag, calling `giu.Update()` every 16 milliseconds is better, but it's still bad if you use it in a popup box

**Comment by Arbureva** [comment]
I have a radio box in the table widget, but whenever I use append() to add it, there are some problems with the radio function, such as two checked cases, but when I reassign it every time, there is no problem
